packages/hardhat/hardhat.config.ts

```diff
@@ -19,6 +19,8 @@ const deployerPrivateKey =
 // If not set, it uses ours Etherscan default API key.
 const etherscanApiKey = process.env.ETHERSCAN_API_KEY || "DNXJA8RX2Q3VZ4URQIWP7Z68CJXQZSC6AW";
 
+const network = process.env.NODE_ENV == "production" ? "optimism" : "baseSepolia";
```

Comment:
Here I believe we should use `VERCEL_ENV` instead of `NODE_ENV`, which changes between `production`, `preview` and `development` as explained in the docs https://vercel.com/docs/projects/environment-variables/system-environment-variables#system-environment-variables
if not, the NODE_ENV variable will be always production because it's a node global variable